services/wfm/src/main/java/org/openkilda/wfm/topology/flow/validation/FlowValidator.java

```diff
@@ -105,9 +111,15 @@ void checkFlowForEndpointConflicts(Flow requestedFlow) throws FlowValidationExce
 
 
         conflictedFlow = conflictsOnDest.stream()
-                .filter(flow -> !requestedFlow.getFlowId().equals(flow.getFlowId())
-                        && (flow.getDestVlan() == 0 || flow.getDestVlan() == requestedFlow.getDestVlan()
-                        || requestedFlow.getDestVlan() == 0))
+                .filter(flow -> !requestedFlow.getFlowId().equals(flow.getFlowId()))
```

Comment:
With these expanded set of conflict tests, the code becomes harder to read. There is no comment that explains what is going on in plain english. I find it is best to include a simple description of what the function does .. ie Endpoint Conflicts .. and in this case, how it does it - ie what are the conditions.